core/src/test/java/io/confluent/rest/TestCustomizedHttpResponseHeaders.java

```diff
@@ -0,0 +1,284 @@
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+import org.junit.runners.Parameterized.Parameter;
+import java.util.Properties;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Configurable;
+import javax.ws.rs.core.MediaType;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+@RunWith(Parameterized.class)
+public class TestCustomizedHttpResponseHeaders {
+
+  private static final Logger log = LoggerFactory.getLogger(TestCustomizeThreadPool.class);
+
+  @Parameter
+  public String headerConfig;
+
+  @Parameters(name = "{0}")
+  public static Object[] data() {
+    return new Object[]{
+            "set",
+            "set add X-XSS-Protection:1",
+            "addX-XSS-Protection",
+            "add X-XSS-Protection:",
+            "X-XSS-Protection:",
+            "add set X-XSS-Protection:",
+            "add X-XSS-Protection:1 X-XSS-Protection:1 ",
+            "set X-Frame-Options:DENY, add  :no-cache, no-store, must-revalidate "
+    };
+  }
+
+  @Test
+  public void testNoCustomizedHeaderConfigs()throws Exception {
+    Properties props = createNoCustomizedHeaderConfigProperties();
+    TestApp app = new TestApp(props);
+    CloseableHttpResponse response = null;
+    try {
+      app.start();
+      response = makeGetRequest(app, "/custom/resource1");
+      String headerValue = getResponseHeader(response, "X-Frame-Options");
+      assertNull(headerValue);
+    } finally {
+      try {
+        if (response != null) {
+          response.close();
+        }
+      } catch (Exception e) {
+      }
+      app.stop();
+    }
+  }
+
+  /**
+   * Testing valid header config
+   * 1) Value has to be in format: [action] [header name]:[header value]. The first part is [action],
+   * the second part is [header name]:[header value] pair seperated by :. There is at least one space between [action]
+   * and [header name]:[header value]
+   * 2) Values for [action] should be in "set, add, setDate, or addDate"
+   */
+  @Test
+  public void testValidHeaderConfigs()throws Exception {
+    Properties props = createValidHeaderConfigProperties();
+    TestApp app = new TestApp(props);
+    CloseableHttpResponse response = null;
+    try {
+      app.start();
+      response = makeGetRequest(app, "/custom/resource1");
+      String headerValue = getResponseHeader(response, "X-Frame-Options");
+      assertEquals("DENY", headerValue);
+      headerValue = getResponseHeader(response, "Cache-Control");
+      assertEquals("no-cache, no-store, must-revalidate", headerValue);
+      headerValue = getResponseHeader(response, "X-Custom-Value");
+      assertNull(headerValue);
```

Comment:
These would be clearer to me if you didn't re-assign the `headerValue` each time:
```
      assertEquals("DENY", getResponseHeader(response, "X-Frame-Options"));
      assertEquals("no-cache, no-store, must-revalidate", getResponseHeader(response, "Cache-Control"));
      assertNull(getResponseHeader(response, "X-Custom-Value"));
```
Or if you added whitespace between the checks:
```
      String headerValue = getResponseHeader(response, "X-Frame-Options");
      assertEquals("DENY", headerValue);

      headerValue = getResponseHeader(response, "Cache-Control");
      assertEquals("no-cache, no-store, must-revalidate", headerValue);

      headerValue = getResponseHeader(response, "X-Custom-Value");
      assertNull(headerValue);
```